File: middleware/src/dsl/Sensor.java
package dsl;

public class Sensor extends SubComponent {
	
	private enum SensorType {
		SONAR,
		GPS_POSITION,
		CAMERA;
	}
	
	private SensorType sensorType = sensorType.SONAR;
	
	public Sensor(SensorType sensor_type) {
		this.sensorType = sensor_type;
	}
	
	public SensorType getType() {
		return sensorType;
	}
	
	public static String sensorTypeToString(SensorType st) {
		return st.name();
	}
}

File: middleware/src/dsl/mission/Sensor.py
from .SubComponent import SubComponent
from enum import Enum

class Sensor(SubComponent):
	
	class SensorType(Enum):
		SONAR = 1,
		GPS_POSITION = 2,
		CAMERA = 3
	
	def __init__(self, id, sensor_type):
		self.sensorType = self.SensorType.sensor_type
		self.ID = id

	def getType(self):
		return self.sensorType
	
	def sensorTypeToString(self, st):
		return st.name()